spec/controllers/people/membership_invoices_controller_spec.rb

```diff
@@ -14,55 +14,128 @@
       person.update!(zip_code: 3600, town: "Thun")
 
-      expect(client).to receive(:create).with(:subject, Hash).and_return({id: 7})
-      expect(client).to receive(:create).with(:address, Hash)
-      expect(client).to receive(:create).with(:communication, Hash)
-      expect(client).to receive(:create).with(:customer, Hash)
-      expect(client).to receive(:create).with(:sales_order, Hash).and_return({sales_order_id: 19})
-      expect(client).to receive(:endpoint).with(:sales_order, Hash)
-      expect(client).to receive(:request).with(:post, String, Hash)
-
       expect do
-        post :create, params: {group_id: groups(:bluemlisalp_mitglieder).id, person_id: person.id, date: "2015-03-01"}
+        post :create, params: {
+          group_id: groups(:bluemlisalp_mitglieder).id,
+          person_id: person.id,
+          people_membership_invoice: {
+            reference_date: Time.zone.today,
+            invoice_date: Time.zone.today,
+            send_date: Time.zone.today,
+            section_id: groups(:bluemlisalp_mitglieder).id,
+            discount: "0"
+          }
+        }
       end.to change { ExternalInvoice.count }.by(1)
 
-      expect(response).to redirect_to(group_person_path(groups(:bluemlisalp_mitglieder).id, person.id))
+      expect(response).to redirect_to(external_invoices_group_person_path(groups(:bluemlisalp_mitglieder).id, person.id))
       expect(flash[:alert]).to be_nil
-      expect(flash[:notice]).to eq("Die Rechnung wurde erfolgreich an Abacus übermittelt. Auftrag-Nr. 19")
+      expect(flash[:notice]).to eq("Die gewünschte Rechnung wird erzeugt und an Abacus übermittelt")
     end
 
-    it "handles failure in abacus request" do
+    it "doesnt create external invoice when params invalid" do
       person.update!(zip_code: 3600, town: "Thun")
 
-      expect(client).to receive(:create).with(:subject, Hash).and_return({id: 7})
-      expect(client).to receive(:create).with(:address, Hash)
-      expect(client).to receive(:create).with(:communication, Hash)
-      expect(client).to receive(:create).with(:customer, Hash)
-      expect(client).to receive(:create).with(:sales_order, Hash).and_raise("Something went wrong")
+      # no send date
+      expect do
+        post :create, params: {
+          group_id: groups(:bluemlisalp_mitglieder).id,
+          person_id: person.id,
+          people_membership_invoice: {
+            reference_date: Time.zone.today,
+            invoice_date: Time.zone.today,
+            send_date: "",
+            section_id: groups(:bluemlisalp_mitglieder).id,
+            discount: "0"
+          }
+        }
+      end.to change { ExternalInvoice.count }.by(0)
+
+      expect(response).to redirect_to(new_group_person_membership_invoice_path(groups(:bluemlisalp_mitglieder).id, person.id))
+      expect(flash[:alert]).to include("Versanddatum muss ausgefüllt werden")
 
+      # no reference date and no invoice date
       expect do
```

Comment:
bitte jeden request in eine eigenes example extrahieren, falls du mehrere von einander abhängige requests machen musst, bitte eine request spec verwenden